code/core/src/core-server/build-index.ts

```diff
@@ -0,0 +1,71 @@
+import { writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { resolve } from 'node:path';
+
+import { loadAllPresets, loadMainConfig, normalizeStories } from '@storybook/core/common';
+import type { BuilderOptions, CLIOptions, LoadOptions } from '@storybook/core/types';
+
+import { logger } from '@storybook/core/node-logger';
+
+import { StoryIndexGenerator } from './utils/StoryIndexGenerator';
+
+type BuildIndexOptions = CLIOptions & LoadOptions & BuilderOptions & { outputFile: string };
+
+export const buildIndex = async (options: BuildIndexOptions) => {
```

Comment:
I think a big chunk of this code could be replaced by a single call to `loadStorybook` - https://github.com/storybookjs/storybook/blob/next/code/core/src/core-server/load.ts/#L15

That function is new in 8.5